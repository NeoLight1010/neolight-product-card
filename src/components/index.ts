import { ProductCard as ProductCardHOC } from './ProductCard';

import { ProductImage } from './ProductImage';
import { ProductTitle } from './ProductTitle';
import { ProductButtons } from './ProductButtons';

export { ProductImage } from './ProductImage';
export { ProductTitle } from './ProductTitle';
export { ProductButtons } from './ProductButtons';

type HOCType = typeof ProductCardHOC;
interface ProductCardHOCInterface extends HOCType {
    Image: typeof ProductImage,
    Title: typeof ProductTitle,
    Buttons: typeof ProductButtons,
}

export const ProductCard: ProductCardHOCInterface = Object.assign(
  ProductCardHOC,
  {
    Image: ProductImage,
    Title: ProductTitle,
    Buttons: ProductButtons,
  }
);

export default ProductCard;
