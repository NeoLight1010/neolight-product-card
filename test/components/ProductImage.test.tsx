import React from "react"; React;

import renderer from "react-test-renderer";
import { ProductCard, ProductImage } from "../../src/components";
import {product2} from "../data/products";

describe("Product Image", () => {
    it("should show component with custom image", () => {
        const wrapper = renderer.create(
            <ProductImage img="./my-custom-image.jpg"/>
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it("should show component with the product's image", () => {
        const wrapper = renderer.create(
            <ProductCard product={product2}>
                {() => (
                    <ProductImage />
                )}
            </ProductCard>
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});
