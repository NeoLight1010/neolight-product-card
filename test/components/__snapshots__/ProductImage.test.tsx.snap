// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Image should show component with custom image 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="./my-custom-image.jpg"
/>
`;

exports[`Product Image should show component with the product's image 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
